examplescripts/gitea_deploy.py

```diff
@@ -0,0 +1,382 @@
+        reservation.data_reservation.networks[0],
+        expiration,
+        None,
+        [reservation],
+        user_form_data["Currency"],
+        config["rid"],
+    )
+    currency = network.currency
+
+    user_form_data["Solution name"] = "gitea_solution_script"
+
+    user_form_data["Public key"] = "Puplic_key".split("\n")[0]
+
+    expiration = j.data.time.get().timestamp + 9900
+    user_form_data["Solution expiration"] = j.data.time.get(expiration).humanize()
+
+    database_name = "postgres"
+    database_user = "postgres"
+    database_password = "postgres"
+    repository_name = "myrepo"
+
+    user_form_data["Database Name"] = database_name
+    user_form_data["Database User"] = database_user
+    user_form_data["Database Password"] = database_password
+    user_form_data["Repository"] = repository_name
+
+    query = {"mru": math.ceil(1024 / 1024), "cru": 2, "sru": 6}
+    # create new reservation
+    reservation = j.sals.zos.reservation_create()
+
+    query["currency"] = currency
+
+    node_selected = j.sals.reservation_chatflow.get_nodes(1, **query)[0]
+    network.add_node(node_selected)
+    ip_address = network.ask_ip_from_node(node_selected)
+    user_form_data["IP Address"] = ip_address
+
+    var_dict = {
+        "pub_key": user_form_data["Public key"],
+        "POSTGRES_DB": user_form_data["Database Name"],
+        "DB_TYPE": "postgres",
+        "DB_HOST": f"{ip_address}:5432",
+        "POSTGRES_USER": user_form_data["Database User"],
+        "APP_NAME": user_form_data["Repository"],
+        "ROOT_URL": f"http://{ip_address}:3000",
+    }
+    database_password_encrypted = j.sals.zos.container.encrypt_secret(
+        node_selected.node_id, user_form_data["Database Password"]
+    )
+    secret_env = {"POSTGRES_PASSWORD": database_password_encrypted}
+    network.update(j.core.identity.me.tid, currency=currency, client=client)
+    storage_url = "zdb://hub.grid.tf:9900"
+    entry_point = "/start_gitea.sh"
+
+    # create container
+    cont = j.sals.zos.container.create(
+        reservation=reservation,
+        node_id=node_selected.node_id,
+        network_name=network.name,
+        ip_address=ip_address,
+        flist=HUB_URL,
+        storage_url=storage_url,
+        env=var_dict,
+        interactive=False,
+        entrypoint=entry_point,
+        cpu=2,
+        public_ipv6=True,
+        memory=1024,
+        secret_env=secret_env,
+    )
+
+    metadata = dict()
+    metadata["chatflow"] = user_form_data["chatflow"]
+    metadata["Solution name"] = user_form_data["Solution name"]
+    metadata["Solution expiration"] = user_form_data["Solution expiration"]
+    metadata["Database name"] = user_form_data["Database Name"]
+    metadata["Database user"] = user_form_data["Database User"]
+    metadata["Database password"] = user_form_data["Database Password"]
+    metadata["Repository"] = user_form_data["Repository"]
+
+    res = j.sals.reservation_chatflow.get_solution_metadata(
```

Comment:
this isn't a chatflow solution right? or is it considered a chatflow solution and so its data should be saved?